b2/account_info/sqlite_account_info.py

```diff
@@ -161,6 +172,29 @@ def _create_tables(self, conn):
            );
         """
         )
+        # Add the 'allowed' column if it hasn't been yet.
+        self._ensure_update(1, 'ALTER TABLE account ADD COLUMN allowed TEXT;')
+
+    def _ensure_update(self, update_number, update_command):
+        """
+        Runs the update with the given number if it hasn't been done yet.
+
+        Does the update and stores the number as a single transaction,
+        so they will always be in sync.
+        """
+        with self._get_connection() as conn:
+            conn.execute('BEGIN')
+            cursor = conn.execute(
```

Comment:
`INSERT OR IGNORE INTO update_done (update_number) VALUES (?);` if the database has a feature for this, we should not be reimplementing it here

Reply:
I could not find that in sqlite.

Reply:
Looks like I missed it.  But I don't think it's what we need here.  We need a transaction that atomically inserts the update number and does the update.  

I stand by the existing implementation.

Reply:
From my reading, the explicit `BEGIN` is necessary for some reason.  The transaction gets committed automatically at the end of the `with` block.